internal/gke/update.go

```diff
@@ -364,6 +364,37 @@ func UpdateMaintenanceWindow(
 	return Changed, nil
 }
 
+// UpdateLabels updates the cluster labels.
+func UpdateLabels(
+	ctx context.Context,
+	client *gkeapi.Service,
+	config *gkev1.GKEClusterConfig,
+	upstreamSpec *gkev1.GKEClusterConfigSpec) (Status, error) {
+	if config.Spec.Labels == nil || reflect.DeepEqual(config.Spec.Labels, upstreamSpec.Labels) || (upstreamSpec.Labels == nil && len(config.Spec.Labels) == 0) {
```

Comment:
nit: I think this is more clear:
```suggestion
	if reflect.DeepEqual(config.Spec.Labels, upstreamSpec.Labels) || (len(upstreamSpec.Labels) == 0 && len(upstreamSpec.Labels) == 0) {
```

Reply:
I don't think that would have quite the same meaning. The three cases I'm going for are
1. Labels is unset by the user and so not owned by the operator, leave it alone no matter what's in upstreamSpec
2. Labels are already matching, no change needed
3. Labels don't technically match, but the upstream labels are unset because there are none and the config spec labels are set to an empty map and so are functionally the same

Combining case 1 and case 3 would ignore the case where there are upstream labels set but we don't want to override them because we don't own them.

Reply:
Yes, that make sense.